src/lib/coda_networking/coda_networking.ml

```diff
@@ -326,6 +326,60 @@ module Make_rpcs (Inputs : Base_inputs_intf) = struct
     end
   end
 
+  module Get_best_tip = struct
+    module Master = struct
+      let name = "get_best_tip"
+
+      module T = struct
+        type query = unit [@@deriving sexp, to_yojson]
+
+        type response =
+          ( External_transition.Stable.V1.t
+          , State_body_hash.Stable.V1.t list * External_transition.Stable.V1.t
+          )
+          Proof_carrying_data.Stable.V1.t
+          option
```

Comment:
> I think it should still returns an option because the peer may not even have a transition frontier because itself is bootstrapping or something.

If you're bootstrapping, you can return your previous root, which would be either the Genesis transition or the transition in persistence.